lib/vfscore/tests/test_vma.c

```diff
@@ -0,0 +1,119 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2025, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <errno.h>
+
+#include <uk/vma_types.h>
+#include <uk/vmem.h>
+#include <uk/test.h>
+#include <uk/list.h>
+#include <uk/print.h>
+#include <uk/alloc.h>
+#include <uk/falloc.h>
+#include <uk/arch/paging.h>
+#include <uk/plat/paging.h>
+#include <uk/nofault.h>
+#include <uk/arch/limits.h>
+
+#include <vfscore/vma.h>
+
+/**
+ * Tests file mappings
+ */
+#define VMEM_TEST_FILENAME "/test_vma_file"
+UK_TESTCASE(ukvmem, test_vma_file)
+{
+	struct uk_vas *vas = vas_init();
+	unsigned char *buf;
+	int fd, i, rc;
+	__vaddr_t va1, va2;
+	__ssz len;
+
+	/* First create a dummy file which we can map. We fill the file with
+	 * pages that start with an incrementing number. This way, we can check
+	 * if we are mapping the correct offset. We also append an incomplete
+	 * page of data at the end, to check if the mapping returns zeroed
+	 * memory for the remainder of the page.
+	 */
+	fd = creat(VMEM_TEST_FILENAME, 0700);
+	vmem_bug_on(fd < 0);
+
+	buf = malloc(PAGE_SIZE);
+	vmem_bug_on(!buf);
+
+	memset(buf, 0xdd, PAGE_SIZE);
+
+	for (i = 0; i < 3; i++) {
+		buf[0] = (char)i;
+
+		len = write(fd, buf, PAGE_SIZE);
+		vmem_bug_on(len != PAGE_SIZE);
+	}
+
+	buf[0] = (char)i;
+	len = write(fd, buf, PAGE_SIZE / 2);
+	vmem_bug_on(len != PAGE_SIZE / 2);
+
+	free(buf);
+
+	/* Create the mapping */
+	va1 = __VADDR_ANY;
+
+	rc = uk_vma_map_file(vas, &va1, 3 * PAGE_SIZE, PROT_R, 0, fd, 0);
+	UK_TEST_EXPECT_ZERO(rc);
+
+	len = probe_r(va1, 3 * PAGE_SIZE);
```

Comment:
I think the build is failing because you are using some function definitions from the original teast_vmem.c